### PR TITLE
Add a BaseOAuth class for Plotly OAuth and others

### PR DESCRIPTION
**Pull request by davecap**
Any thoughts on something like this? It would be helpful for us to enable OAuth on our Dash apps (at SolveBio).

### CODE REVIEW

**Comment by chriddyp** [comment]
seems reasonable to me!

**Comment by davecap** [comment]
Great, is there anything you want me to change? What should I do to get this merged in?

File: dash_auth/oauth.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import datetime
+import flask
+from flask_seasurf import SeaSurf
+import json
+import os
+from .auth import Auth
+
+
+class OAuthBase(Auth):
+    AUTH_COOKIE_NAME = 'plotly_auth'
```
Comment (akhmerov) [inline dash_auth/oauth.py]: Should this be `plotly_auth` in the base class?

Comment (davecap) [reply dash_auth/oauth.py]: How about something like `dash_token`?

**Comment by akhmerov** [comment]
Sounds reasonable.

**Comment by chriddyp** [comment]
I'll take another look at this tomorrow and then merge 👍

**Comment by davecap** [comment]
Bump!

**Comment by chriddyp** [comment]
This is great. Thanks for doing this @davecap and @akhmerov !

**Pull request merged**